Config added: Per server/library configuration example and instructions. (#1814)
Taxel/PlexTraktSync: README.md

INFO     Enabled 2 libraries in Plex Server: ['Movies', 'TV Shows']
```

### Per server configuration

If you want to specify your config per server you can do so inside of `servers.yml`. Within the `config` part of the server configuration you can specify how that specific server should work.

```yml
  Example1:
    token: ~
    urls:
      - http://localhost:32400
    config:
      sync:
        plex_to_trakt:
          collection: true
        trakt_to_plex:
          liked_lists: false
```

> Both `watch` and `sync` can be used inside of the config key-value.

Reviewer: no. only sync is supported. where you took watch is supported too?

Author: I think read it somewhere, but can't find that specific info I read. It could be that I misunderstood #1206 thinking it meant that watch was supported. I'll update it later today.

> Only `sync` can be used inside of the config key-value. Keep in mind that the sync-config set in `servers.yml` overrides the global sync-config in `config.yml`.

Reviewer: "only sync" is not true either, because other section describes "library" and "excluded-libraries".

perhaps reword as: "sync" section overrides can be used or something like that.

Author: That is very true. Nice catch. What do you think about: "Using `sync` in a server config overrides the global sync-config in `config.yml`." as an alternative?

Author: Commited an alternative! Let me know.

If you want to run these jobs using `ofelia`, you can do so by running something similar to this in your `docker-compose.yml`:
```yml
  plextraktsync:
    image: ghcr.io/taxel/plextraktsync
    command: sync
    container_name: plextraktsync
    profiles: ["schedule"]
    volumes:
      - /configs/mediarr/plextraktsync:/app/config
    environment:
      - PUID=1000
      - PGID=1000
    depends_on:
      - plex
  scheduler:
    image: mcuadros/ofelia:latest
    container_name: scheduler
    command: daemon --docker
    restart: unless-stopped
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    labels:
      ofelia.job-run.plextraktsync.schedule: "0 6,18 * * *"
      ofelia.job-run.plextraktsync.container: "plextraktsync"
      ofelia.job-run.plextraktsync.command: "--server 'Example1' sync"
      ofelia.job-run.plextraktsync2.schedule: "0 12,0 * * *"
      ofelia.job-run.plextraktsync2.container: "plextraktsync"
      ofelia.job-run.plextraktsync2.command: "--server 'Example2' sync"
```

> This means that a job is running every 6 hours, alternating between the two "servers". Keep in mind that since this example assumes that you are only running one container, you need to make sure that the jobs don't overlap.

Reviewer: why jobs can't overlap?

Author: Because if the job takes a long time (say over half a second) Ofelia doesn't want to create the same container again. Maybe there is a cleaner way? Not super used to Ofelia.

Reviewer: so, that just means it runs only one job at a time.

either remove the warning, or explain what it's for.

INFO     Enabled 2 libraries in Plex Server: ['Movies', 'TV Shows']
```

### Per server configuration

If you want to specify your config per server you can do so inside of `servers.yml`. Within the `config` part of the server configuration you can specify how that specific server should work.

```yml
  Example1:
    token: ~
    urls:
      - http://localhost:32400
    config:
      sync:
        plex_to_trakt:
          collection: true
        trakt_to_plex:
          liked_lists: false
```
Using `sync` in a server config overrides the global sync-config in `config.yml`.

> This can also be used to have different configs between different libraries. To be able to do this you specifiy the number of servers you need (most likely equal to the number of different config setups you need). For example:

Reviewer: ```suggestion
This can also be used to have different configs between different libraries. To be able to do this you specify the number of servers you need (most likely equal to the number of different config setups you need). For example:
```

Author: LET ME TELL YOu. I misspelled specify so many times last night... I guess you caught one of the ones I missed.

```yml
  Example1:
    token: ~
    urls:
      - http://localhost:32400
    config:
      libraries:
        - "Movies"
      sync:
        plex_to_trakt:
          ratings: true
          watched_status: true
        trakt_to_plex:
          ratings: true
          watched_status: true
  Example2:
    token: ~
    urls:
      - http://localhost:32400
    config:
      libraries:
        - "TV Shows"
      sync:
        plex_to_trakt:
          ratings: true
          watched_status: false
        trakt_to_plex:
          ratings: true
          watched_status: false
```
The above config would make it so that the "Movies" library syncs both ratings and watched status, while the "TV Shows" library only syncs ratings.

> To then run the sync you need to specify `--server Example1` or `--server Example2` to run the sync for that specific server. Running the sync command without it will result in running the sync using the default config.

Reviewer: ```suggestion
To then run the sync you need to specify `--server Example1` or `--server Example2` to run the sync for that specific server.

Running the sync command without `--server` will use default server from `.env`
```

again. not true statement detected. --server doesn't specify what configs to override, but rather what server to pick. your default server could be either of those examples or something completely different.

Author: Yea! sorry.
Didnt know that default server were saved somewhere.

If you want to run these jobs using `ofelia`, you can do so by running something similar to this in your `docker-compose.yml`:
```yml
  plextraktsync:
    image: ghcr.io/taxel/plextraktsync
    command: sync
    container_name: plextraktsync
    profiles: ["schedule"]
    volumes:
      - /configs/mediarr/plextraktsync:/app/config
    environment:
      - PUID=1000
      - PGID=1000
    depends_on:
      - plex
  scheduler:
    image: mcuadros/ofelia:latest
    container_name: scheduler
    command: daemon --docker
    restart: unless-stopped
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    labels:
      ofelia.job-run.plextraktsync.schedule: "0 6,18 * * *"
      ofelia.job-run.plextraktsync.container: "plextraktsync"
      ofelia.job-run.plextraktsync.command: "--server 'Example1' sync"
      ofelia.job-run.plextraktsync2.schedule: "0 12,0 * * *"
      ofelia.job-run.plextraktsync2.container: "plextraktsync"
      ofelia.job-run.plextraktsync2.command: "--server 'Example2' sync"
```

> If you are running only one PlexTraktSync container, you need to make sure that the two jobs Ofelia jobs don't run at the same time. Depending on how long it takes for the job to run on your server you might have to keep the schedule of the two jobs seperated with a few minutes or a few hours. If you have two different PlexTraktSync containers in your docker compose, you can run them at the same time.

Reviewer: ```suggestion
If you are running only one PlexTraktSync container, you need to make sure that the two jobs Ofelia jobs don't run at the same time. Ofelia skips scheduling a new job run if the previous job is still running.

  Depending on how long it takes for the job to run on your server you might have to keep the schedule of the two jobs seperated with a few minutes or a few hours. If you have two different PlexTraktSync containers in your docker compose, you can run them at the same time.
```

you still failed to describe the reasoning why overlapping must be avoided.

Author: lol true.

> The above config means that a job is running every 6 hours, alternating between the two "servers". The PlexTraktSync container also has a profile (docker-compose feature) called "schedule" which means that it won't run automatically when you run for example `docker-compose up`.

Reviewer: ```suggestion
The above config means that a job is running every 6 hours, alternating between the two "servers". The PlexTraktSync container also has a [docker compose profile] called "schedule" which means that it won't run automatically when you run for example `docker-compose up`.

[docker compose profile]: https://docs.docker.com/compose/profiles/
```

add link to doc

Author: Good idea!
